ironfish/src/network/peerNetwork.test.ts

```diff
@@ -198,6 +200,43 @@ describe('PeerNetwork', () => {
       expect(peerNetwork['node']['syncer'].addNewBlock).toHaveBeenCalledWith(peer, block)
     })
 
+    describe('handle block gossip', () => {
+      const nodeTest = createNodeTest()
+
+      it('should mark block hashes as known and known on peers', async () => {
+        const { strategy, chain, peerNetwork, syncer } = nodeTest
+
+        const genesis = await chain.getBlock(chain.genesis)
+        Assert.isNotNull(genesis)
+
+        strategy.disableMiningReward()
+        syncer.blocksPerMessage = 1
+
+        const blockA1 = await makeBlockAfter(chain, genesis)
+
+        const { peer: peer1 } = getConnectedPeer(peerNetwork.peerManager)
+        const { peer: peer2 } = getConnectedPeer(peerNetwork.peerManager)
+        const { peer: peer3 } = getConnectedPeer(peerNetwork.peerManager)
+        peer1.knownPeers.set(peer2.getIdentityOrThrow(), peer2)
+        peer2.knownPeers.set(peer1.getIdentityOrThrow(), peer1)
+
+        const newBlockMessage = new NewBlockMessage(strategy.blockSerde.serialize(blockA1))
+
+        const peer1Send = jest.spyOn(peer1, 'send')
+        const peer2Send = jest.spyOn(peer2, 'send')
+        const peer3Send = jest.spyOn(peer3, 'send')
+
+        await peerNetwork['handleGossipMessage'](peer1, newBlockMessage)
```

Comment:
curious - is there a reason we call the method this way as opposed to `peerNetwork.handleGossipMessage`?

Reply:
It's a private method, so we can't call it directly, but typescript lets us sneak around the privacy this way. I'll double check and see if there's a public method we can call.